src/Compilers/Core/Portable/PEWriter/ManagedResource.cs

```diff
@@ -48,7 +48,11 @@ public void WriteData(BlobBuilder resourceWriter)
                         var count = (int)(stream.Length - stream.Position);
                         resourceWriter.WriteInt32(count);
 
-                        resourceWriter.WriteBytes(stream, count);
+                        int bytesWritten = resourceWriter.TryWriteBytes(stream, count);
+                        if (bytesWritten != count)
+                        {
+                            throw new EndOfStreamException($"Resource stream ended at {bytesWritten} bytes, expected {count} bytes");
```

Comment:
They have to be localizable. However, if they end up incorporated into a diagnostic, it is OK to use CurrentUICulture as the target language for localization and not use the culture passed in to DiagnosticInfo.GetMessage.